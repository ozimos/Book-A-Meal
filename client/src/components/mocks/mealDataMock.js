export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
