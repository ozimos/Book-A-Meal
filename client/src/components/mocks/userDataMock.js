export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' }
};
export const userIncludes = {
  email: 'asdf',
  firstName: 'first',
  lastName: 'last'
};
