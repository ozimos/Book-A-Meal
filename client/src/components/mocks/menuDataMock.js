import { allMeals } from './mealDataMock';

export default {
  data: { Meals: allMeals.data }
};