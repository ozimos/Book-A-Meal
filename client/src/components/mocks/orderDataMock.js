import { userIncludes } from './userDataMock';
import { rows } from './mealDataMock';

const Meals = rows.map(meal =>
  ({ ...meal, MealOrders: { quantity: 2 } }));

const template = {
  id: 'orderId',
  updatedAt: '2018-08-24T16:47:55.783Z',
  userId: 'userId',
  Meals,
  User: userIncludes,
};

export const order = {
  data: {
    pages: 1,
    offset: 0,
    rows: [template]
  }
};

export const orderActionData = {
  order: template

};

export const allOrders = {
  data: {
    rows: [{ ...template, id: 'order1' },
      { ...template, id: 'order2' }] }
};
