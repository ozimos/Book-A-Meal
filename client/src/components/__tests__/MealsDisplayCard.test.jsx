/* global React:false, shallow:false toJson:false */

import MealsDisplayCard from '../mealCard/MealDisplayCard';
import { meal } from '../mocks/mealDataMock';

const props = {
  ...meal,
  collection: 'menu'
};

describe('MealsDisplayCard Component', () => {
  const setup = () => shallow(<MealsDisplayCard {...props} />);
  it('renders correctly', () => {
    let wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
    const props2 = { ...props, addToCollection: jest.fn() };
    const setup2 = () => shallow(<MealsDisplayCard {...props2} />);
    wrapper = setup2();
    expect(toJson(wrapper)).toMatchSnapshot();
    const props3 = { ...props, removeFromCollection: jest.fn() };
    const setup3 = () => shallow(<MealsDisplayCard {...props3} />);
    wrapper = setup3();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

