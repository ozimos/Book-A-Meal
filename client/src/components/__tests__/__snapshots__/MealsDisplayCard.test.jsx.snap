// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealsDisplayCard Component renders correctly 1`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          />
          <span>
            ₦
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
      />
    </div>
    <p
      className="descrip long_string"
    />
  </div>
</React.Fragment>
`;

exports[`MealsDisplayCard Component renders correctly 2`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          />
          <span>
            ₦
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
      />
    </div>
    <p
      className="descrip long_string"
    />
    <button
      className="btn title-button"
      onClick={[Function]}
    >
      Add to menu
    </button>
  </div>
</React.Fragment>
`;

exports[`MealsDisplayCard Component renders correctly 3`] = `
<React.Fragment>
  <div
    className="card"
  >
    <div
      className="overlay-container"
    >
      <div
        className="overlay"
      >
        <div
          className="flexbox"
        >
          <span
            className="long_string"
          />
          <span>
            ₦
          </span>
        </div>
      </div>
      <img
        alt="Meal"
        className="fluid-img"
      />
    </div>
    <p
      className="descrip long_string"
    />
    <button
      className="btn title-button"
      onClick={[Function]}
    >
      Remove from menu
    </button>
  </div>
</React.Fragment>
`;
